lib/writable.js

```diff
@@ -238,6 +238,7 @@ class Writable extends EventEmitter {
     }
 
     if (chunk) this.write(chunk, encoding, callback);
+    else if (callback) this.once('finish', callback);
```

Comment:
The callback must always be called with `'finish'` event, even when the `chunk` is provided.